opensourceleg/actuators/base.py

```diff
@@ -0,0 +1,423 @@
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Set,
+    TypeVar,
+    Union,
+    cast,
+    runtime_checkable,
+)
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from functools import partial
+
+import numpy as np
+
+from opensourceleg.logging.logger import LOGGER
+
+
+class MOTOR_CONSTANTS(NamedTuple):
+    # TODO: Add thermal constants
+    MOTOR_COUNT_PER_REV: float
+    NM_PER_AMP: float
+    NM_PER_RAD_TO_K: float
+    NM_S_PER_RAD_TO_B: float
+
+    MAX_CASE_TEMPERATURE: float
+    MAX_WINDING_TEMPERATURE: float
+
+    @property
+    def RAD_PER_COUNT(self) -> float:
+        if self.MOTOR_COUNT_PER_REV == 0:
+            return 0
+
+        return 2 * np.pi / self.MOTOR_COUNT_PER_REV
+
+    @property
+    def NM_PER_MILLIAMP(self) -> float:
+        return self.NM_PER_AMP / 1000
+
+
+class CONTROL_MODES(Enum):
+    POSITION = 0
+    CURRENT = 1
+    VOLTAGE = 2
+    IMPEDANCE = 3
+    VELOCITY = 4
+    TORQUE = 5
+    IDLE = 6
+
+
+@dataclass(kw_only=True)
+class ControlGains:
```

Comment:
This seems like a dephy specific setup, not general actuators. I would not include it in the base class.

Reply:
Agreed! So far we've been trying to fit classes that interface with other actuators to fall into this pattern, which is probably not great. For example, actuators that don't support impedance control (Moteus) still has "k" and "b" terms.

What would be a general way to do this without creating multiple sub-classes? Will refactoring ControlGains to be a NamedTuple instead of a Dataclass be a good change? This way we can default values to None and this would match the CONTROL_MODE_CONFIGS setup.